builtin/providers/ultradns/resource_ultradns_dirpool.go

```diff
@@ -0,0 +1,612 @@
+									},
+								},
+							},
+						},
+						"ip_info": &schema.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"is_account_level": &schema.Schema{
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
+									},
+									"ips": &schema.Schema{
+										Type:     schema.TypeSet,
+										Optional: true,
+										Set:      hashIPInfoIPs,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"start": &schema.Schema{
+													Type:     schema.TypeString,
+													Optional: true,
+													// ConflictsWith: []string{"cidr", "address"},
+												},
+												"end": &schema.Schema{
+													Type:     schema.TypeString,
+													Optional: true,
+													// ConflictsWith: []string{"cidr", "address"},
+												},
+												"cidr": &schema.Schema{
+													Type:     schema.TypeString,
+													Optional: true,
+													// ConflictsWith: []string{"start", "end", "address"},
+												},
+												"address": &schema.Schema{
+													Type:     schema.TypeString,
+													Optional: true,
+													// ConflictsWith: []string{"start", "end", "cidr"},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			// Computed
+			"hostname": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// CRUD Operations
+
+func resourceUltradnsDirpoolCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*udnssdk.Client)
+
+	r, err := makeDirpoolRRSetResource(d)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] ultradns_dirpool create: %#v", r)
+	_, err = client.RRSets.Create(r.RRSetKey(), r.RRSet())
+	if err != nil {
+		// FIXME: remove the json from log
+		marshalled, _ := json.Marshal(r)
+		ms := string(marshalled)
+		return fmt.Errorf("create failed: %#v [[[[ %v ]]]] -> %v", r, ms, err)
+	}
+
+	d.SetId(r.ID())
+	log.Printf("[INFO] ultradns_dirpool.id: %v", d.Id())
+
+	return resourceUltradnsDirpoolRead(d, meta)
+}
+
+func resourceUltradnsDirpoolRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*udnssdk.Client)
+
+	rr, err := makeDirpoolRRSetResource(d)
+	if err != nil {
+		return err
+	}
+
+	rrsets, err := client.RRSets.Select(rr.RRSetKey())
+	if err != nil {
+		uderr, ok := err.(*udnssdk.ErrorResponseList)
+		if ok {
+			for _, resps := range uderr.Responses {
+				// 70002 means Records Not Found
+				if resps.ErrorCode == 70002 {
+					d.SetId("")
+					return nil
+				}
+				return fmt.Errorf("resource not found: %v", err)
+			}
+		}
+		return fmt.Errorf("resource not found: %v", err)
+	}
+
+	r := rrsets[0]
```

Comment:
Are we sure that the record at index 0 is the one we want? Shouldn't we include some check in the range above to make sure?